### PR TITLE
Verify signatures on transaction acceptence.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/crypto.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
 use parity_scale_codec::{Decode, Encode};
 use std::fmt::{self, Debug, Formatter};
+use ursa::{
+    keys::PublicKey as UrsaPublicKey,
+    signatures::{ed25519::Ed25519Sha512, SignatureScheme},
+};
```
Comment (kamilsa) [inline iroha/src/crypto.rs:6]: In one of the PRs Egor introduced Crypto module. Would be great to reuse it here as well

File: iroha/src/tx.rs [hunk 2]
```diff
@@ -84,6 +84,11 @@ impl Transaction {
             signatures,
         } = self
         {
+            for signature in &signatures {
+                if signature.verify(&Vec::from(&request)).is_err() {
```
Comment (kamilsa) [inline iroha/src/tx.rs:88]: Message that we verify should be scale encoded transaction. Is it the case here?

File: iroha/src/crypto.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/crypto.rs:6]: @kamilsa I think you talk about https://github.com/hyperledger/iroha/pull/465 - this is the same module (look at the file name)

File: iroha/src/tx.rs (hunk 2 above)
Comment (humb1t) [reply iroha/src/tx.rs:88]: @kamilsa yes - we do not expose `scale` to have an ability to replace with any other binary format. right now we use `Into\From for Vec<u8>` and inside this impls we use `scale`

**Pull request merged**